### DISCUSSION TITLE
Delete the out directory if it already exists when exporting

### DISCUSSION THREAD

**Discussion by bennyburrito** [resend/react-email#1371] [Ideas]
### Goals

1. keeps the out directory up to date
2. removes duplicate emails being created
3.

### Background

When I have been creating emails, if I change the folder structure, I then have to remember to delete the out directory before exporting again otherwise the export creates duplicate files which then gets confusing. Simply deleting the export directory of choice if it already exists means it is only creating whatever is in the folders at the time and removes old templates. We use this system on our current setup and it is very useful. 

### Proposal

By adding this code below under the spinner, emailsDirectoryMetadata... variable at the top of the exportTemplates function variable in the node_modules/react-email/cli/index.js file, when someone runs the email export command, it will delete whatever directory they have chosen or the out as default if the folder already exists. I have tested it and it works as expected and means I can trust the emails in the folder and correct at the time of exporting.

// Delete the out directory if it exists
        if (import_node_fs4.default.existsSync(pathToWhereEmailMarkupShouldBeDumped)) {
            import_shelljs.rm('-rf', pathToWhereEmailMarkupShouldBeDumped);
        }

**Comment by gabrielmfern** [comment]
What version of `react-email` are you using here? IIRC we had a fix for this, not sure if on the latest or on canary though.

**Reply by bennyburrito** [reply]
2.1.0-canary.1

**Reply by gabrielmfern** [reply]
Oh ok, I misread what you mentioned; your issue is that there are stale files after modifying the file structure.
Would love a PR on this as it seems you already have some idea of how to fix it. If not, opening an issue would be lovely to track this
as I would consider this as an issue here.

**Reply by gabrielmfern** [reply]
For the record, I was thinking of duplicate static files not email templates.

https://github.com/resend/react-email/blob/4439ac2a9255189466040ab5986edb62d8823bbd/packages/react-email/src/cli/commands/export.ts#L126-L129

**Reply by bennyburrito** [reply]
No worries. Consider it requested.
https://github.com/resend/react-email/pull/1373

**Reply by gabrielmfern** [reply]
Thank you :D! Appreciate your time on this very much.
